src/boot/efi/vmm.c

```diff
@@ -12,12 +12,82 @@
 #include "util.h"
 #include "vmm.h"
 
+/* Keep CVM detection logic in this file at feature parity with
+ * that in src/basic/confidential-virt.c */
```

Comment:
can this be shared instead of being duplicated, at least in part? We already share code between uefi mode and userspace, see under src/fundamental/

Reply:
If we can conditionalize the MSR handling, we could likely share it here. Am I right that ``src/fundamental`` will be compiled twice, with ``#if SD_BOOT`` being the condition to check for whether we're built in EFI context ?

Reply:
yep

Reply:
maybe just make the msr handling a func pointer param, and then we pass one implementation in in efi mode, and another in userspace. Sounds simple enough, and this is not performance sensitive, so the indirection really doesn't matter